File: backend/internal/buses/bus_model.go
package buses

import "time"

type Bus struct {
	ID         string    `json:"id"`
	BusName    string    `json:"bus_name"`
	BusNumber  string    `json:"bus_number"`
	RouteID    string    `json:"route_id"`
	TotalSeats int       `json:"total_seats"`
	CreatedAt  time.Time `json:"created_at"`
}

type CreateBusRequest struct {
	BusName    string `json:"bus_name"`
	BusNumber  string `json:"bus_number"`
	RouteID    string `json:"route_id"`
	TotalSeats int    `json:"total_seats"`
}

File: backend/frontend/lib/models/bus_model.dart
class BusModel {
  final String id;
  final String busName;
  final String busNumber;
  final int totalSeats;

  BusModel({
    required this.id,
    required this.busName,
    required this.busNumber,
    required this.totalSeats,
  });

  factory BusModel.fromJson(Map<String, dynamic> json) {
    return BusModel(
      id: json['id'] ?? "",

      busName: json['bus_name'] ?? "",

      busNumber: json['bus_number'] ?? "",

      totalSeats: json['total_seats'] ?? 0,
    );
  }
}
